src/query/users/src/role_mgr.rs

```diff
@@ -52,6 +67,55 @@ impl UserApiProvider {
         Ok(roles)
     }
 
+    /// Ensure the builtin roles account_admin and public exists. Please note that there's
+    /// a corner case that if we added another privilege type, we should add it to the
+    /// existed account_admin role.
+    ///
+    /// This function have two calling places:
+    /// 1. when the server starts
+    /// 2. when the user is authenticated
+    #[async_backtrace::framed]
+    pub async fn ensure_builtin_roles(&self, tenant: &str) -> Result<()> {
```

Comment:
maybe we can add a ut about this? revoke a exists privilege in account_admin like SUPER. then call this function, the account_admin's privilege will contain SUPER

Reply:
makes sense, let me add it

Reply:
i've added a ut about this, PTAL